python/lsst/ts/observatory/control/maintel/mtcalsys.py

```diff
@@ -0,0 +1,567 @@
+            intended_usage=intended_usage,
+        )
+
+        self.mtcamera = mtcamera
+        self.ls_select_led_location = 0.0  # um
+        self.ls_select_laser_location = 20.0  # um
+
+        self.laser_enclosure_temp = 20.0  # C
+
+        # ALl this needs to go in a config yaml file somewhere.
+        self.led_per_filter = {
+            "u": ["M385L3"],
+            "g": ["M455L4", "M505L4"],
+            "r": ["M565L3", "M660L4"],
+            "i": ["M730L5", "M780LP1"],
+            "z": ["M850L3", "M940L3"],
+            "y": ["M970L4"],
+        }
+        self.ls_led_locations = {
+            "u": 0.0,
+            "g": 1.0,
+            "r": 2.0,
+            "i": 3.0,
+            "z": 4.0,
+            "y": 5.0,
+        }
+        self.ls_led_focus = {"u": 0.0, "g": 1.0, "r": 2.0, "i": 3.0, "z": 4.0, "y": 5.0}
+
+        self.exptime_dict: dict[str, float] = dict(
+            camera=0.0,
+            electrometer=0.0,
+            fiberspectrograph=0.0,
+        )
+        self.delay = 1
+
+    def ls_laser_focus(self, wavelength: float) -> float:
+        """Function for evaluating the distance of the focus linear stage"""
+        return 10.0
+
+    async def change_wavelength(self, wavelength: float) -> None:
+        """Change the TunableLaser wavelength setting"""
+
+        task_wavelength = self.rem.tunablelaser.cmd_changeWavelength.set_start(
+            wavelength=wavelength, timeout=self.long_long_timeout
+        )
+        task_focus = self.linearstage_laser_focus.cmd_moveAbsolute.set_start(
+            distance=self.ls_laser_focus(wavelength)
+        )
+
+        await asyncio.gather(task_wavelength, task_focus)
+
+    async def is_ready_for_flats(self) -> bool:
+        """Add doctring"""
+        # TODO (DM-44310): Implement method to check that the
+        # system is ready for flats.
+        return True
+
+    async def setup_calsys(self, sequence_name: str) -> None:
+        """If monochromatic flats, check that laser can be enabled,
+        check temperature, turn on laser to warm up.
+        move linearstage_select to correct location for
+        Mono or Whitelight flats
+        """
+        config_data = self.get_calibration_configuration(sequence_name)
+
+        calibration_type = getattr(CalibrationType, str(config_data["calib_type"]))
+
+        if calibration_type == CalibrationType.WhiteLight:
+            await self.linearstage_projector_select.cmd_moveAbsolute.set_start(
+                distance=self.ls_select_led_location
+            )
+        else:
+            await self.linearstage_projector_select.cmd_moveAbsolute.set_start(
+                distance=self.ls_select_laser_location
+            )
+            await self.laser_setup(config_data["laser_mode"])
+            await self.rem.tunablelaser.cmd_startPropagating.set_start()
+
+    async def laser_setup(self, mode: str) -> None:
+        """Is the laser on? what is the temp like? thermal control on?"""
```

Comment:
This docustring is a bit weird, can you please rephrase it? also, please document the input parameter.